src/main/java/com/cloudbees/jenkins/plugins/bitbucket/BitbucketBuildStatusNotifications.java

```diff
@@ -63,14 +65,29 @@ private static String getRootURL(@NonNull Run<?, ?> build) {
         }
 
         String url = DisplayURLProvider.get().getRunURL(build);
+        return checkURL(url);
+    }
 
-        if (url.startsWith("http://localhost")) {
-            throw new IllegalStateException("Jenkins URL cannot start with http://localhost");
-        }
+    /**
+     * Check if the build URL is compatible with Bitbucket API.
+     * For example, Bitbucket API doesn't accept simple hostnames as URLs host value
+     * Throws an IllegalStateException if it is not valid, or return the url otherwise
+     *
+     * @param url the URL of the build to check
+     * @return the url if it is valid
+     */
+    static String checkURL(@NonNull String url) {
         if (url.equals("http://unconfigured-jenkins-location/")) {
```

Comment:
Actually looking at blue ocean display API it should be `url.contains("http://unconfigured-jenkins-location/")`

https://github.com/jenkinsci/blueocean-display-url-plugin/blob/d6515117712c9b30a32f32bb6e1a977711b9b037/src/main/java/org/jenkinsci/plugins/blueoceandisplayurl/BlueOceanDisplayURLImpl.java#L57-L63